src/mongo_odm/test/query_builder.cpp

```diff
@@ -127,6 +128,7 @@ TEST_CASE("Test nested member access.", "[mongo_odm::nvp_child]") {
     }
 
     SECTION("Nested member access using MONGO_ODM_CHILD macro") {
+        // Nvp's must be used as temporary objects.
```

Comment:
The NvpChild object stores parent Nvp's by reference, and the other expressions / nvp types do this as well. Which means that if one is stored in a variable, and then used later on, it's `Parent` member is now invalid. IIRC @hanumantmk  mentioned that if we store everything by value, when the expressions get built, successive copies of sub-expressions might end up blowing the stack. 

It might be worth having the Nvp types store their members by value, since they won't get so big, and just leaving the expressions to store members by reference which would make these tests clearer.